From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Finn Kuenstner <finn.kuenstner@gmail.com>
Date: Wed, 8 Mar 2023 19:57:51 +0100
Subject: [PATCH] Don't print "Couldn't load server icon" NPE twice


diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index bbb8335dae0a3e2761e6bbb8dc723bcf28cd82ba..181319182b8fb9bd7f8504f9432b07bfd0a41fc2 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -1065,7 +1065,7 @@ public final class CraftServer implements Server {
                 this.icon = CraftServer.loadServerIcon0(file);
             }
         } catch (Exception ex) {
-            this.getLogger().log(Level.WARNING, "Couldn't load server icon", ex);
+            //this.getLogger().log(Level.WARNING, "Couldn't load server icon", ex); // Paper - Don't print "Couldn't load server icon" NPE twice
         }
     }
 
